### DISCUSSION TITLE
Hanger resonator design+sim

### DISCUSSION THREAD

**Discussion by Qvadi** [iqm-finland/KQCircuits#64] [Q&A]
Hi KQC,
I have been trying to design a few hanger resonators (capacitively coupled) to a feedline. I have a couple of questions:

1. I wonder if there are scripts to automatically estimate resonance frequencies of the resonator lengths (end to end nodes - combination of waveguide coplanar+meander) based on analytical models? 
2. Is there a more recent script to export the design to HFSS to simulate the eigen modes? I am using Ansys 2023 R1. The one that is currently there crashes even if I disable elmer in the code. The ansys engine is not started.

**Comment by qpavsmi** [comment]
I can answer the second point to some extent. If you have saved your design as python class that is a subclass of `Element`, you can use the `get_single_element_sim_class` to generate a simulation class that we can export as a simulation script that can be used for external simulators, either Elmer or Ansys. See tutorial [here ](https://iqm-finland.github.io/KQCircuits/user_guide/simulation/simulation_object.html) and also consult example simulation scripts we have under [simulation scripts](https://github.com/iqm-finland/KQCircuits/tree/main/klayout_package/python/scripts/simulations). But you seem to have issues with having Ansys work with the exported files, so maybe if you would paste the code you wrote to export simulation files to be used for Ansys we can see what could be the issue.

I will try to find someone who could help you with the first point.
